src/main/java/luke/Main.java

```diff
@@ -0,0 +1,149 @@
+import luke.task.Task;
+import luke.task.Todo;
+
+
+public class Main extends Application {
+    public static void main(String[] args) {
+        File tlFile = new File("./data/", "luke.txt");
+        Scanner sc = new Scanner(System.in);
+        TaskList taskList = new TaskList();
+        taskList.readTasks(tlFile);
+        System.out.printf("Luke:\n\tHey there! I'm Luke.\n\tPlease tell me what to add to your list.\nYou:\n");
+        while (true) {
+            String input = sc.nextLine().toLowerCase();
+            if (Pattern.matches("^(todo) *.*$", input)) {
+                try {
+                    Todo newTodo = createTodo(input);
+                    taskList.writeTasks(tlFile, newTodo);
+                } catch (EmptyTodoException e) {
+                    System.out.printf("Luke:%s\nYou:\n", e.getMessage());
+                }
+            } else if (Pattern.matches("^(deadline) *.*$", input)){
+                try {
+                    Deadline newDeadline = createDeadline(input);
+                    taskList.writeTasks(tlFile, newDeadline);
+                } catch (EmptyDeadlineException e) {
+                    System.out.printf("Luke:%s\nYou:\n", e.getMessßage());
+                } catch (InvalidDeadlineException e) {
+                    System.out.printf("Luke:%s\nYou:\n", e.getMessage());
+                }
+            } else if (Pattern.matches("^(event) *.*$", input)) {
+                try {
+                    Event newEvent = createEvent(input);
+                    taskList.writeTasks(tlFile, newEvent);
+                } catch (EmptyEventException e) {
+                    System.out.printf("Luke:%s\nYou:\n", e.getMessage());
+                } catch (InvalidEventException e) {
+                    System.out.printf("Luke:%s\nYou:\n", e.getMessage());
+                }
+            } else if (input.equals("list")) {
+                if (taskList.getSize() < 1) {
+                    System.out.printf("Luke:\n\tYou don't have any tasks in your list :(\nYou:\n");
+                } else {
+                    String todoSummary = "Luke:\n\tHere are the tasks in your list.";
+                    for (int i = 0; i < taskList.getSize(); i++) {
+                        Task current = taskList.getTask(i);
+                        todoSummary += String.format("\n\t%d.%s", i + 1, current);
+                    }
+                    System.out.printf("%s\nYou:\n", todoSummary);
+                }
+            } else if (Pattern.matches("^(done) *[0-9]*$", input)) {
+               try {
+                   taskList.completeTask(input);
+               } catch (InvalidDoneException e) {
+                   System.out.printf("Luke:%s\nYou:\n", e.getMessage());
+               } catch (DoneIndexOutofboundsException e) {
+                   System.out.printf("Luke:%s\nYou:\n", e.getMessage());
+               }
+            } else if (Pattern.matches("^(delete) *[0-9]*$", input)) {
+                try {
```

Comment:
Your try-catch blocks all look very similar. Perhaps you can factor this out into a function (and put it in the `Command` class mentioned above).

Reply:
Nice suggestion. Seems like the implementation of function could really make the codes more concise.